### ISSUE TITLE
Index Page - 001

### ISSUE THREAD

**Issue by Wmaxlees** [amc-ucdenver/website#13]
As a student, I would like to see upcoming events on the index page

**Comment by robfitzgerald** [comment]
taking a swing at describing it as BDD:

Scenario 1: Index page has an Upcoming Events section
Given the internet isn't broken
When the user points their browser to our URL
Then we should be able to tell that we are at the ACM of UCD site by it's title
  and we should be able to find an upcoming events section

Scenario 2: See Upcoming Events
Given the user is on the index page
  and the database contains valid events objects
When we look at the content in the subset of the page containing upcoming events
Then upcoming events should be listed
  and it should match the content in the database

Scenario 3: Upcoming Events Pagination
Given the user is on the index page
When we look at the content in the subset of the page containing upcoming events
Then upcoming events should be listed
  and there should be a pager at the bottom of the section
  and clicking the pager should show the next subset of upcoming events

**Comment by Wmaxlees** [comment]
Hmmm. I think that's pretty close.

Maybe:

Scenario 1: Index page shows upcoming events
   **Given** there are upcoming events
   **When** the user navigates to the index page
   **Then** the user should see upcoming events section
   **And** the upcoming events should match the event in the database

Scenario 2: There are no upcoming events
   **Given** there are no upcoming events
   **When** the user navigates to the index page
   **Then** the user should see an empty upcoming events section

I don't think the title has anything to do with this user story. If you are worried about that, maybe start a new one along the lines of "As a user, I want to see a title on every page so I know I am on the UCDenver ACM website."

**Comment by robfitzgerald** [comment]
:+1:

**Comment by robfitzgerald** [comment]
revisiting this description of the test.  if it is an end-to-end test, it shouldn't be validating the events against the database - that's part of the scope of index's unit test.
- e2e for this should be seeing if the index page has an events section or not.
- unit for this should be testing the factory, the controller, then the view, and confirming that nothing is corrupt, and that it has sound business logic.

Scenario 1: Index page shows upcoming events section
Given we direct the browser to a page with the upcoming events section
When the page is fully loaded
Then the user should see upcoming events section
And that section should be a table (or any other design requirements...)

Scenario 2: Upcoming Events displays events
Given we direct the browser to a page with the upcoming events section
When there are upcoming events
Then the user should see all display fields of an upcoming event displayed

**Comment by robfitzgerald** [comment]
just a stab at the e2e test.  kinda rushed it, any thoughts/edits welcome!

**Comment by robfitzgerald** [comment]
maybe it should also have a 

When there are no upcoming events
Then there is text displayed saying "there are no upcoming events"

**Comment by Wmaxlees** [comment]
Yeah I like what you have so far if you include the "no upcoming events" scenario. Let me think about it a bit. We may be able to come up with some more scenarios

**Comment by robfitzgerald** [comment]
ok if you like it i'll start with this.  adding more would be juustttt fiiiine

**Comment by robfitzgerald** [comment]
back after a totally KILLER leftover reuben from yesterday's lunch at "jason's deli" (mind blown)...

Scenario 1: Index page shows upcoming events section
Given we direct the browser to a page with the upcoming events section
When the page is fully loaded
Then the user should see upcoming events section
And that section should be a table (or any other design requirements...)

Scenario 2: Upcoming Events displays events
Given we direct the browser to a page with the upcoming events section
When there are upcoming events
Then the user should see all display fields of an upcoming event displayed

Scenario 3: No Upcoming Events message
Given we direct the browser to a page with the upcoming events section
When there aren't any upcoming events
Then the user should see the message "There are no upcoming events."

**Comment by robfitzgerald** [comment]
... scenario 4: how to handle http errors?  maybe.  maybe trivial, since if the API is failing for this, then, it's likely that the whole server would be failing.

**Comment by Wmaxlees** [comment]
Yeah, I don't think we need scenario 4. That should be handled by the unit testing?

**Comment by robfitzgerald** [comment]
sure.
